keras_cv/layers/preprocessing/repeated_augmentation.py

```diff
@@ -0,0 +1,113 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer,
+)
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class RepeatedAugmentation(BaseImageAugmentationLayer):
+    """RepeatedAugmentation augments each image in a batch multiple times.
+
+    This technique exists to emulate the behavior of stochastic gradient descent within
+    the context of mini-batch gradient descent.  When training large vision models,
+    choosing a large batch size can introduce too much noise into aggregated gradients
+    causing the overall batch's gradients to be less effective than gradients produced
+    using smaller gradients.  RepeatedAugmentation handles this by re-using the same
+    image multiple times within a batch creating correlated samples.
+
+    This layer increases your batch size by a factor of `len(augmenters)`.
+
+    Args:
+        augmenters: the augmenters to use to augment the image
+        shuffle: whether or not to shuffle the result.  Essential when using an
+            asynchronous distribution strategy such as ParameterServerStrategy.
+
+    Usage:
+
+    List of identical augmenters:
+    ```python
+    repeated_augment = cv_layers.RepeatedAugmentation(
+        augmenters=[cv_layers.RandAugment(value_range=(0, 255))] * 8
+    )
+    inputs = {
+        "images": tf.ones((8, 512, 512, 3)),
+        "labels": tf.ones((8,)),
+    }
+    outputs = repeated_augment(inputs)
+    # outputs now has a batch size of 64 because there are 8 augmenters
+    ```
+
+    List of distinct augmenters:
+    ```python
+    repeated_augment = cv_layers.RepeatedAugmentation(
+        augmenters=[
+            cv_layers.RandAugment(value_range=(0, 255)),
+            cv_layers.RandomFlip(),
+        ]
+    )
+    inputs = {
+        "images": tf.ones((8, 512, 512, 3)),
+        "labels": tf.ones((8,)),
+    }
+    outputs = repeated_augment(inputs)
+    ```
+
+    References:
+    - [DEIT implementaton](https://github.com/facebookresearch/deit/blob/ee8893c8063f6937fec7096e47ba324c206e22b9/samplers.py#L8)
+    - [Original publication](https://openaccess.thecvf.com/content_CVPR_2020/papers/Hoffer_Augment_Your_Batch_Improving_Generalization_Through_Instance_Repetition_CVPR_2020_paper.pdf)
+
+    """
+
+    def __init__(self, augmenters, shuffle=True, **kwargs):
+        super().__init__(**kwargs)
+        self.augmenters = augmenters
+        self.shuffle = shuffle
+
+    def _batch_augment(self, inputs):
+        if 'bounding_boxes' in inputs:
+            raise ValueError(
+                "RepeatedAugmentation() does not yet support bounding box labels."
+            )
+
+        augmenter_outputs = [augmenter(inputs) for augmenter in self.augmenters]
```

Comment:
Ok here eventually we have the opportunity to tell to the augmenter to not randomize the augmentation param within the batch.

Reply:
Check my comment at https://github.com/keras-team/keras-cv/pull/1293#discussion_r1083533391